src/util/animate.ts

```diff
@@ -32,7 +32,7 @@ import { Point } from '../point.class';
    * @type {AnimationContext[]}
    */
   var fabric = global.fabric, RUNNING_ANIMATIONS = [];
-  fabric.util.object.extend(RUNNING_ANIMATIONS, {
+  extend(RUNNING_ANIMATIONS, {
```

Comment:
Doesn't have to be a subclass of array, it can be a class that has an array referenced. It doesn't really matter too much to me. Here i just swapped fabric.util.object because i had issues with circular imports

Reply:
even better